### ISSUE TITLE
Do not apply new-module-imports rule to an addon's app folder

### ISSUE THREAD

**Issue by Dhaulagiri** [ember-cli/eslint-plugin-ember#190]
When updating addons to use the new modules syntax we have to avoid updating any files in the addon's `app/` directory since that is transpiled by the addon's consuming app which may not have an up-to-date version of ember-cli-babel.  With this addon installed though eslint will fail any files in the app directory that violate this rule.  I added a one file exception to the affected file in my addon for now, but I'm not sure if there's a better way to handle this.

**Comment by rwjblue** [comment]
We should add an override for an addons `app/**/*.js` for this rule.

**Comment by Dhaulagiri** [comment]
This may be wishful thinking but at this point it'd be nice if this rule enforced that the new modules syntax _is not_ used in this directory as I've been bitten by this once already.  That may be a better thing to transfer over to the codemods repo though since I think it is being overeager in this case (at least for now).

**Comment by rwjblue** [comment]
I don’t think that we can do anything here for that. I was referring to the default setup for addons (which will include this plugin in their eslint config as of ember-cli@2.18) in my comment above. 

I totally agree that the codemod should be better and do a more intelligent thing based on the type of project. This was exactly what I was referring to in: https://github.com/ember-cli/ember-modules-codemod/issues/31.
